src/database/Dockerfile.local

```diff
@@ -0,0 +1,35 @@
+FROM postgres:12.17
+
+ENV POSTGRES_USER=opensupplyhub \
+    POSTGRES_PASSWORD=opensupplyhub \
+    POSTGRES_DB=opensupplyhub
+
+
+# Build and install the PostGIS package from scratch to ensure it matches the
+# LLVM version used by the Docker container based on postgres:12.17 image.
+# See PostGIS installation details at https://postgis.net/docs/manual-3.4/postgis_installation.html
+
+# Install the requirements for building and using PostGIS.
+RUN apt-get update && apt-get install -y \
+    wget \
+    tar \
+    build-essential \
+    libproj-dev \
+    libgeos-dev \
+    libjson-c-dev \
+    libgdal-dev \
+    libprotobuf-c-dev \
+    protobuf-c-compiler \
+    pkg-config \
+    postgresql-server-dev-12
+
+# Download the PostGIS source archive, build and install the PostGIS package.
+RUN wget http://download.osgeo.org/postgis/source/postgis-3.4.0.tar.gz && \
+    tar -xvzf postgis-3.4.0.tar.gz &&  \
+    cd /postgis-3.4.0 && \
+    ./configure && \
+    make && \
+    make install && \
+    cd .. && \
+    rm -rf postgis-3.4.0 && \
+    rm postgis-3.4.0.tar.gz
```

Comment:
Combine most of the commands into one layer to build and install the PostGIS package, and then remove artifacts like the source folder. This helps decrease the size of the image.